Data/NF/Internal.hs

```diff
@@ -17,11 +21,17 @@
 -- Stability   :  experimental
 -- Portability :  portable
 --
--- This module provides the constructor for the type 'NF', allowing
--- you to create values of type 'NF' without any runtime overhead
--- if you fulfill a proof obligation that the value is already in
--- normal form.
-module Data.NF.Internal(NF(..)) where
+-- This module is intended for internal use.
+module Data.NF.Internal(
+#if __GLASGOW_HASKELL__ >= 800
+  NF(..,NF)
+#else
+  NF(..)
+#if __GLASGOW_HASKELL__ >= 710
+  , pattern NF
```

Comment:
I don't understand. I care more about offering the best API possible than
about trying to make users write portable code. If their code breaks on
some other implementation, that's because they chose the convenience I
offered; they didn't have to.

On Jul 22, 2016 6:17 PM, "Edward Z. Yang" <notifications@github.com> wrote:

> In Data/NF/Internal.hs
> <https://github.com/ezyang/nf/pull/6#discussion_r71953969>:
>
> > @@ -17,11 +21,17 @@
> >  -- Stability   :  experimental
> >  -- Portability :  portable
> >  --
> > --- This module provides the constructor for the type 'NF', allowing
> > --- you to create values of type 'NF' without any runtime overhead
> > --- if you fulfill a proof obligation that the value is already in
> > --- normal form.
> > -module Data.NF.Internal(NF(..)) where
> > +-- This module is intended for internal use.
> > +module Data.NF.Internal(
> > +#if __GLASGOW_HASKELL__ >= 800
> > +  NF(..,NF)
> > +#else
> > +  NF(..)
> > +#if __GLASGOW_HASKELL__ >= 710
> > +  , pattern NF
>
> I guess the point is, giving people NF encourages non-portable behavior.
> It's easy enough to not use it, so let's not make it easy.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ezyang/nf/pull/6/files/db57fa061540b46fb54928e8131e84be376b7fe9#r71953969>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABzi_R1CoJYIkCuIlYmIf1EnbKjMjntJks5qYUGJgaJpZM4JTKLc>
> .
>